src/rez/shells.py

```diff
@@ -160,6 +160,29 @@ def spawn_shell(self, context_file, tmpdir, rcfile=None, norc=False,
         """
         raise NotImplementedError
 
+    @classmethod
+    def get_key_token(cls, key, form=0):
```

Comment:
I find the use of `get_key_token`/`key_form_count` a bit confusing. How about we just update `get_key_token` to return a list, where the first item is the preferred form? It's not used in many places so not much code would need updating, and I'd be pretty surprised if anyone considered this part of the public API (so I don't expect backwards compatibility to be an issue here).

Reply:
To add to this - the func should be allowed to return str or list of str, so that any existing external shell plugins won't break. So in that respect we need backwards compatibility, but I think that will be enough.

Reply:
Did it as I proposed. This is way cleaner, the API stays compatible and the naming is way clearer.
Hope you agree.